File: src/main/java/io/github/trainb0y/simplehud/elements/BiomeElement.java
package io.github.trainb0y.simplehud.elements;

import net.minecraft.client.MinecraftClient;
import net.minecraft.util.registry.Registry;
import net.minecraft.world.biome.Biome;

import java.util.List;

// Hud element displaying the player's current biome
public class BiomeElement extends Element {
    public BiomeElement() {
        super();
    }

    public BiomeElement(int x, int y, boolean enabled) {
        super(x, y, enabled);
    }

    public Object[] getArgs(MinecraftClient client) {
        Biome biome = client.world != null ? client.world.getBiome(client.player != null ? client.player.getBlockPos() : null).value() : null;
        String name = String.valueOf(client.world != null ? client.world.getRegistryManager().get(Registry.BIOME_KEY).getId(biome) : null);
        try {
            name = name.split(":")[1].replace("_", " ");
        } catch (IndexOutOfBoundsException ignored) {
        }

        return List.of(name).toArray(); // formatted biome name
    }

    @Override
    public String getKey() {
        return "element.simplehud.biome";
    }
}

File: src/main/kotlin/io/github/trainb0y/fabrihud/elements/BiomeElement.kt
package io.github.trainb0y.fabrihud.elements

import net.minecraft.client.MinecraftClient
import net.minecraft.util.registry.Registry

// Hud element displaying the player's current biome
class BiomeElement : TextElement() {
	override fun getArgs(client: MinecraftClient): List<Any?> {
		val biome = client.world?.getBiome(client.player?.blockPos)?.value()
		var name = client.world?.registryManager?.get(Registry.BIOME_KEY)?.getId(biome).toString()
		try {
			name = name.split(":")[1].replace("_", " ")
		} catch (ignored: IndexOutOfBoundsException) {
		}
		return listOf(name) // formatted biome name
	}

	override val key = "element.fabrihud.biome"
}